packages/remix-dev/config/routes.ts

```diff
@@ -137,11 +144,13 @@ export function defineRoutes(
       options = optionsOrChildren || {};
     }
 
+    let fileId = createRouteId(file);
+
     let route: ConfigRoute = {
       path: path ? path : undefined,
       index: options.index ? true : undefined,
       caseSensitive: options.caseSensitive ? true : undefined,
-      id: createRouteId(file),
+      id: options.id ? `${fileId}--${options.id}` : fileId,
```

Comment:
> what is the best way to "verbatim and error if they provide duplicates" just throws a new Error inside routes.ts file?

Maybe just change this to 

```suggestion
      id: options.id || createRouteId(file),
```

And delete line 147 above